packages/compass-import-export/src/export/export-json.ts

```diff
@@ -0,0 +1,129 @@
+import { Transform } from 'stream';
+import { pipeline } from 'stream/promises';
+import { EJSON } from 'bson';
+import type { Writable } from 'stream';
+import toNS from 'mongodb-ns';
+import type { DataService } from 'mongodb-data-service';
+
+import { createReadableCollectionCursor } from '../utils/collection-stream';
+import type {
+  ExportQuery,
+  ExportAggregation,
+} from '../utils/collection-stream';
+import { createDebug } from '../utils/logger';
+
+const debug = createDebug('export-json');
+
+type ExportJSONOptions = {
+  dataService: DataService;
+  ns: string;
+  output: Writable;
+  abortSignal: AbortSignal;
+  query?: ExportQuery;
+  aggregation?: ExportAggregation;
+  progressCallback: (index: number) => void;
```

Comment:
I'd make progressCallback and abortSignal optional to keep in line with where import ended up.